File: 1168-duplicate-zeros/duplicate-zeros.cpp
class Solution {
public:
    void duplicateZeros(vector<int>& arr) {
        int n  = arr.size();
        int zeroes = 0;
        vector<int> new_arr(n,0);
        for (int i = 0;i<n;i++){
            if (arr[i] == 0){
                zeroes++;
            }
            else{
                if (i+zeroes < n)
                new_arr[i+zeroes] = arr[i];
            }
        }
        for (int i = 0;i<n;i++){
            arr[i] = new_arr[i];
        }
        // return new_arr;
        
    }
};

File: 1168-duplicate-zeros/duplicate-zeros.py
class Solution:
    def duplicateZeros(self, arr: List[int]) -> None:
        """
        Do not return anything, modify arr in-place instead.
        """
        
        n = len(arr)
        zeroes = 0
        new_arr = [0] * n
        
        for i in range(n):
            if arr[i] == 0:
                zeroes += 1
                continue
            if i + zeroes < n:
                new_arr[i + zeroes] = arr[i]
            else:
                break

        for i in range(n):
            arr[i] = new_arr[i]
